salt/minion.py

```diff
@@ -157,8 +158,13 @@ def resolve_dns(opts, fallback=True):
                 True,
                 opts['ipv6'])
         except SaltClientError:
+            retry_dns_count = opts.get('retry_dns_count', None)
             if opts['retry_dns']:
                 while True:
+                    if retry_dns_count is not None:
+                        if retry_dns_count == 0:
+                            raise SaltMasterUnresolvableError
+                        retry_dns_count -= 1
```

Comment:
Well, that's the thing. I'd set it default to something big enough. Any `None/0` or negative is "until forever".

Reply:
This looks correct to me. It should default "until forever" because that is the current behavior which we don't want to break for people.